hivemind/p2p/p2p_daemon.py

```diff
@@ -504,8 +481,22 @@ def _convert_process_arg_type(val: Any) -> Any:
     def _maddrs_to_str(maddrs: List[Multiaddr]) -> str:
         return ",".join(str(addr) for addr in maddrs)
 
+    async def _read_outputs(self, ready: asyncio.Future) -> None:
+        last_line = None
+        while True:
+            line = await self._child.stdout.readline()
+            if not line:  # Stream closed
+                break
+            last_line = line.rstrip().decode(errors="ignore")
+
+            if last_line.startswith("Peer ID:"):
+                ready.set_result(None)
+
+        if not ready.done():
+            ready.set_exception(P2PDaemonError(f"Daemon failed to start: {last_line}"))
```

Comment:
As it appears, P2P.create will hang if daemon does not print the desired line but does not stop either. If so, it would be great to introduce some kind of timeout for ready.set_result and consider daemon failed if it doesn't succeed in time